core/src/test/scala/unit/kafka/admin/TopicCommandTest.scala

```diff
@@ -188,4 +189,103 @@ class TopicCommandTest extends ZooKeeperTestHarness with Logging with RackAwareT
     }
     checkReplicaDistribution(assignment, rackInfo, rackInfo.size, alteredNumPartitions, replicationFactor)
   }
+
+  def testDescribePartitions() {
```

Comment:
The tests are testing two things here. Copying from my previous comment.

1. Unit tests to test that special reports conditions are valid. For instance, for a given topic under-replicated or unavailable partitions are detected correctly.
2. As these cli tools are the only way, as of now, to discover these info. Ops folks may have automation scripts around these tools that parse tools' output to get various info. Having checks to know when the output is changing will avoid unintentional changes becoming a pain for such ops guys. I personally do not know of any ops guys doing this, but this is what I have heard from multiple people.